src/compliance.rs

```diff
@@ -0,0 +1,171 @@
+use anyhow::{Context, Result};
+use bitcoin::Address;
+use chrono::prelude::*;
+use fancy_regex::Regex;
+use futures::StreamExt;
+use log::{error, info};
+use std::{
+    collections::HashMap,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+use tokio::{spawn, sync::RwLock, task::JoinHandle, time::interval};
+use xml::reader::{EventReader, XmlEvent};
+
+pub struct Compliance {
+    sanctioned_addresses: Arc<RwLock<HashMap<String, bool>>>,
+    last_update: Arc<RwLock<i64>>,
+}
+
+impl Default for Compliance {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```

Comment:
weird :D

Reply:
oops. I blindly followed clippy suggestion. Will change this to Derive macro